src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java

```diff
@@ -292,6 +297,14 @@ private String extractGroupId(String appId) {
 		return groupId;
 	}
 
+	private Collection<Constraint> deduceConstraints(AppDeploymentRequest request) {
```

Comment:
The method `deduceConstraints` can possibly refactored to use `AppDeploymentRequest` and `DeploymentProperty key` and thereby can be used by both the app deployer and chronos task launcher.